### DISCUSSION TITLE
Implements Artifacts with Local Ollama models

### DISCUSSION THREAD

**Discussion by CIDE-Jim** [danny-avila/LibreChat#4446] [Feature Requests & Suggestions]
The artifacts interface does not appear when using a local Ollama model. Is it possible to extend artifacts to be available to local models as well?

**Comment by Exioness** [comment]
Artifacts only work with LLM models that are capable of processing them, which requires the ability to follow complex instructions. Therefore, smaller models (e.g., 2B, 3B parameters) typically won't be able to handle artifacts effectively, as they lack the capacity to follow instructions in this way.

**Reply by CIDE-Jim** [reply]
Thanks for the clarification. I appreciate your insight.

**Reply by sanjeevkumarraob** [reply]
I am using llama 3.1 8b parameter model with my Ollama. I am getting the code suggestions in a regular code block but not in artifacts panel. Is there any configuration to be provided in the Ollama custom endpoint?

**Reply by danny-avila** [reply]
If you haven't seen the video, it explains how this feature works and how it can be used: https://www.youtube.com/watch?v=GfTj7O4gmd0

**Reply by sanjeevkumarraob** [reply]
Yeah. I have seen the video and pretty impressed about the functionality. at one point, the prompt prefix includes a custom definition for artifacts. But I can't find these information anywhere in the documentation. Is it possible to add more details about generating artifacts in the docs?

<img width="636" alt="Screenshot 2024-10-20 at 10 36 29 AM" src="https://github.com/user-attachments/assets/79aad213-1cc2-4391-baa6-f5e146aba3af">
